integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -762,6 +771,30 @@ private int rmInternal(String path, boolean mustBeFile) {
     return 0;
   }
 
+  /**
+   * Waits for the file to complete before opening it.
+   *
+   * @param uri the file path to check
+   * @return whether the file is completed or not
+   */
+  private boolean waitForFileCompleted(AlluxioURI uri) {
+    try {
+      CommonUtils.waitFor("file completed", () -> {
+        try {
+          return mFileSystem.getStatus(uri).isCompleted();
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }, WaitForOptions.defaults().setTimeoutMs(MAX_OPEN_WAITTIME));
+      return true;
+    } catch (InterruptedException ie) {
+      Thread.currentThread().interrupt();
+      return false;
+    } catch (TimeoutException te) {
+      return false;
+    }
+  }
```

Comment:
If it is not too much trouble, could you put the FUSE changes in a separate PR? These seem like independent changes.

Reply:
Sure, I would put Fuse changes in another PR, thanks!